client/src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react';
+import ModalOption from './ModalOption';
+/* 
+  props: {
+    type: 'success' | 'warning' | 'failure' | 'neutral',
+    title: string,
+    description: string,
+    options: Array<{
+      label: string,
+      action: (e: Event) => void,
+    }>
+    onDisplay: (v: bool) => void
+    show: bool
+  }
+*/
+
+const Modal = ({ type, title, description, options, onDisplay, show }) => {
+  const [display, setDisplay] = useState(false);
+
+  const classList = ['modal', !show ? 'modal--hidden' : null];
```

Comment:
nie wiem jak to dziala, ale pusty string tj '' wydaje się bezpieczniejszy przy joinowaniu niż null. Tym bardziej, że jak przejdziesz na inne języki niż js to taka konstrukcja da Ci nulla ;)

Ja raczej zawartość class list bym walnął bezpośrednio niżej w returnie.

Wyliczania nazwy klasy w zależności od typu dałbym w oddzielnej funkcji

Reply:
Faktycznie, nie zauważyłem tego. Już to poprawiłem.